sdk/python/kubeflow/storage_init_container/hugging_face.py

```diff
@@ -57,3 +59,37 @@ def download_model_and_tokenizer(self):
         transformers.AutoTokenizer.from_pretrained(
             self.model, cache_dir=self.config.download_dir
         )
+
+
+@dataclass
+class HfDatasetParams:
+    repo_id: str
+    access_token: str = None
+    allow_patterns: list[str] = None
+    ignore_patterns: list[str] = None
+    download_dir: str = field(default="/workspace/datasets")
+
+    def __post_init__(self):
+        # Custom checks or validations can be added here
+        if self.repo_id is None:
+            raise ValueError("repo_id is None")
+
+
+class HuggingFaceDataset(datasetProvider):
+    def load_config(self, serialised_args):
+        self.config = HfDatasetParams(**json.loads(serialised_args))
+
+    def download_dataset(self):
+        print("downloading dataset")
+        import huggingface_hub
+        from huggingface_hub import snapshot_download
+
+        if self.config.access_token:
+            huggingface_hub.login(self.config.access_token)
+        snapshot_download(
+            repo_id=self.config.repo_id,
+            repo_type="dataset",
```

Comment:
Can we use upstream constants?

https://github.com/huggingface/huggingface_hub/blob/2effe3f9d0fc47c8c6a934b321f5a2551f330caf/src/huggingface_hub/constants.py#L73-L76

Reply:
removed this piece of code